uri.go

```diff
@@ -29,46 +41,41 @@ var (
 // DNS hostname instead.
 //
 // See: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
-//
-var SchemesWithDNSHost map[string]bool
-
-func init() {
-	SchemesWithDNSHost = map[string]bool{
-		"dns":      true,
-		"dntp":     true,
-		"finger":   true,
-		"ftp":      true,
-		"git":      true,
-		"http":     true,
-		"https":    true,
-		"imap":     true,
-		"irc":      true,
-		"jms":      true,
-		"mailto":   true,
-		"nfs":      true,
-		"nntp":     true,
-		"ntp":      true,
-		"postgres": true,
-		"redis":    true,
-		"rmi":      true,
-		"rtsp":     true,
-		"rsync":    true,
-		"sftp":     true,
-		"skype":    true,
-		"smtp":     true,
-		"snmp":     true,
-		"soap":     true,
-		"ssh":      true,
-		"steam":    true,
-		"svn":      true,
-		"tcp":      true,
-		"telnet":   true,
-		"udp":      true,
-		"vnc":      true,
-		"wais":     true,
-		"ws":       true,
-		"wss":      true,
-	}
+var SchemesWithDNSHost = map[string]bool{
```

Comment:
I considered using `map[string]struct{}` here as that would avoid the unnecessary `bool` value given that all values are true (and thus decrease the memory usage a bit). However, that would be a breaking change so I backed that out. I can always add it in if you want me to though.

Reply:
Actually, it would be even better to just use a switch case. It will be over 3x faster because the compiler can optimize it.